### ISSUE TITLE
Compiler Error C7595 with VS 2019 16.10 Preview 2.0

### ISSUE THREAD

**Issue by Ryan-rsm-McKenzie** [marzer/tomlplusplus#93]
## Environment
**toml++ version and/or commit hash:**  2.3.0 / [4face4d5bf16326aca0da1fb33876dbca63b6e2f](https://github.com/marzer/tomlplusplus/tree/4face4d5bf16326aca0da1fb33876dbca63b6e2f)

**Compiler:**  VS 2019 16.10 Preview 2.0

**C++ standard mode (e.g. 17, 20, 'latest'):**  C++20

**Target arch (e.g. x64):**  x64

**Library configuration overrides:**  None

**Relevant compilation flags:**  `/c++latest`

## Describe the bug
VS errors on compilation with the message:
`include\toml++\toml_parser.hpp(478,41): error C7595: 'toml::v2::operator ""_sz': call to immediate function is not a constant expression`

## Steps to reproduce (or a small repro code sample)
I'm no language expert, but you can see from this simplified [godbolt example](https://godbolt.org/z/8cvnf61GT) that GCC and MSVC currently disagree over whether this is valid or not.

## Additional information
Here's the offending line:
https://github.com/marzer/tomlplusplus/blob/7a4993029e3f49ea42a12cea252755b440c530f8/include/toml%2B%2B/toml_parser.hpp#L478
And the custom operator. Note that it defers to `consteval`. MSVC would happily compile this if you used `constexpr` instead.
https://github.com/marzer/tomlplusplus/blob/7a4993029e3f49ea42a12cea252755b440c530f8/include/toml%2B%2B/toml_common.h#L692

**Comment by marzer** [comment]
That operator is probably the most elementary example of a trivially-consteval function I can think of. This is almost certainly a bug in the new MSVC. Thanks for letting me know!

**Comment by marzer** [comment]
@Ryan-rsm-McKenzie I've pushed a fix for this in master. Since this is very likely a bug specific to the preview version's implementation of `consteval` I've fixed it for this very specific `_MSC_FULL_VER` in the hopes that it is fixed by the time VS 16.10.0 comes out of preview.

I've filed a compiler bug report [here](https://developercommunity.visualstudio.com/t/Erroneous-C7595-error-with-consteval-in/1404234), consider upvoting it to help it along!
